### ISSUE TITLE
StackMobs incorrectly handling cancelled events.

### ISSUE THREAD

**Issue by JamieSinn** [Nathat23/StackMob-2#54]
Brief Description: When someone, not part of the island comes on and tries to shear Mooshrooms they duplicate.
Instructions:
Try to shear Mooshrooms on an island you aren't part of.

How many times did you recreate this?:
10+ times.
Result:
The stack of 100 Mooshrooms duplicates into another stack of 100.
Expected Result:
I'd assume they would do what sheep do if you try to shear them without being added to the island, one unstacks and simply re-stack in a few seconds.

Basically, any event that causes an entity change (and is cancelled) is not properly handled on StackMobs's side it seems (though, this could be an issue on both yours, and ASkyBlock's side)

@andrewkm can confirm.

**Comment by andrewkm** [comment]
Can confirm this here.

**Comment by andrewkm** [comment]
@Nathat23 can confirm that this is now handled correctly with other plugins.
Tested and confirmed. Thank you!

Definitely a very big fix for the plugin :)
